### PR TITLE
test: Fix NodeLogQuery tests

### PR DESCRIPTION
**Pull request by aravindhp**
#### What type of PR is this?
/kind bug
/kind failing-test

#### What this PR does / why we need it:
This PR fixes the NodeLogQuery test.

- Remove redundant tests
- Fix formatting of the query command by using fmt.Sprintf to prevent spurious characters from being introduced
- Fix running of the journalctl command on the node by add the default options
- Restrict running the tests on a single node

### CODE REVIEW

File: test/e2e/node/kubelet.go [hunk 1]
```diff
@@ -458,187 +458,144 @@ var _ = SIGDescribe("kubelet", func() {
...
 		*/
 
-		ginkgo.It("should return the error of running without --query option", func() {
+		ginkgo.It("should return the error with an empty --query option", func() {
 			ginkgo.By("Starting the command")
 			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
 
-			for nodeName := range nodeNames {
-				queryCommand := "\"/api/v1/nodes/" + nodeName + "/proxy/logs/?"
+			for nodeName := range linuxNodeNames {
```
Comment (marosset) [inline test/e2e/node/kubelet.go:489]: Do we need to query all the linux nodes?

File: test/e2e/node/kubelet.go [hunk 2]
```diff
@@ -458,187 +458,144 @@ var _ = SIGDescribe("kubelet", func() {
...
 		/*
-			Test if kubectl get --raw "/api/v1/nodes/<insert-node-name-here>/proxy/logs/?query=kubelet&boot=-1"
-			returns something or not!
+			Test if kubectl get --raw "/api/v1/nodes/<insert-node-name-here>/proxy/logs/?query=kubelet&tailLines=3"
+			returns the last three lines of the kubelet log
 		*/
 
-		ginkgo.It("should return the kubelet logs for the previous boot", func(ctx context.Context) {
+		ginkgo.It("should return the last three lines of the kubelet logs", func(ctx context.Context) {
+			ginkgo.Skip("FIXED")
```
Comment (marosset) [inline test/e2e/node/kubelet.go:543]: did you mean to leave the `ginkgo.SKip("FIXED")` in the PR?

File: test/e2e/node/kubelet.go (hunk 1 above)
Comment (aravindhp) [reply test/e2e/node/kubelet.go:489]: I think testing one node is sufficient but I was just fixing the test. I am fine just picking one node and using that.

File: test/e2e/node/kubelet.go (hunk 2 above)
Comment (aravindhp) [reply test/e2e/node/kubelet.go]: Sorry about that. Fixed.

File: test/e2e/node/kubelet.go (hunk 1 above)
Comment (marosset) [reply test/e2e/node/kubelet.go:489]: We can do that in a separate PR too

**Review by marosset — APPROVED** [review]
/lgtm

**Comment by kannon92** [comment]
Linter looks to be a real issue.

**Comment by aravindhp** [comment]
@kannon92 I fixed the issue revealed by the linter-hints job by getting rid of the Set which also addresses @marosset's [comment](https://github.com/kubernetes/kubernetes/pull/121707#discussion_r1380878568).

File: test/e2e/node/kubelet.go (hunk 1 above)
Comment (aravindhp) [reply test/e2e/node/kubelet.go]: Given the linter-hints job was complaining about the Set, I restricted the test to one Linux node  in this PR itself.

**Review by mrunalp — APPROVED** [review]

**Pull request merged**